packages/overlay/src/Overlay.ts

```diff
@@ -565,10 +565,7 @@ export class Overlay extends OverlayFeatures {
                 `${this.tagName.toLowerCase()}-${randomID()}`
             );
         }
-        if (
-            changes.has('open') &&
-            (typeof changes.get('open') !== 'undefined' || this.open)
-        ) {
+        if (changes.has('open') && (this.hasUpdated || this.open)) {
```

Comment:
if we already trying to find 'open' attributes from the PropertyValues do we also need a check on `this.open` ?

Reply:
This specific says, when `open` has changed, but NOT when open has changed from `undefined` (its initial value) to `false` (its default value). That was the if only returns true when it's not the first update or the end user applied `open === true` in the DOM of the element before it was upgraded.